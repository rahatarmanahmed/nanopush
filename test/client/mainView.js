const test = require('ava')
const window = require('global/window')

const mainView = require('../../client/mainView')
const { TOKEN } = require('../helpers/testData')

test.before(() => {
  window.location = {
    origin: 'http://localhost:3000'
  }
})

test.after(() => {
  delete window.location
})

test('mainView redirects to / if token not set', t => {
  const html = mainView({}, null, (action, data) => {
    t.is(action, 'location:set')
    t.is(data, '/')
  })

  t.snapshot(html.toString())
})

test('mainView renders all the docs and stuff with a given token', t => {
  const html = mainView({ token: TOKEN })

  t.snapshot(html.toString())
})
