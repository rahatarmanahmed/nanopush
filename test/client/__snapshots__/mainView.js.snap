exports[`mainView redirects to / if token not set 1`] = `"\"<div></div>\""`;

exports[`mainView renders all the docs and stuff with a given token 1`] = `"\"<div class=\\\"measure\\\">\\n      <section class=\\\"mv4\\\">\\n        <p class=\\\"lh-copy\\\">\\n          Your push notification URL is:\\n        </p>\\n        <div class=\\\"code pa2 br2 bg-dark-green light-blue overflow-y-scroll nowrap\\\">\\n          http://localhost:3000/h/3766ff2f-c295-481e-8b78-03d8e04cfc56/notify\\n        </div>\\n\\n        <p class=\\\"lh-copy\\\">\\n          You can receive notifications when you send a request to that URL, even when this page isn\'t open. This URL will stay the same (even if you refresh) until you clear your browser data. If you want a new token, <a class=\\\"link underline hover-light-blue pointer\\\">click here</a>.\\n        </p>\\n      </section>\\n\\n      <div class=\\\"nowrap overflow-hidden tc\\\">ðŸ“ŒðŸ“ŒðŸ“ŒðŸ“ŒðŸ“Œ</div>\\n\\n      <section class=\\\"mv4\\\">\\n        <p class=\\\"lh-copy\\\">\\n          Try it out! Run this command to see if it works\\n        </p>\\n        <div class=\\\"code pa2 br2 bg-dark-green light-blue\\\">\\n          curl \'http://localhost:3000/h/3766ff2f-c295-481e-8b78-03d8e04cfc56/notify?title=nanopush&amp;body=This%20is%20a%20test\'\\n        </div>\\n      </section>\\n\\n      <div class=\\\"nowrap overflow-hidden tc _f3841cc1\\\">ðŸ“ŒðŸ“ŒðŸ“ŒðŸ“ŒðŸ“Œ</div>\\n\\n      <section class=\\\"mv4\\\">\\n        <h2 class=\\\"f4 fw7 mb2 lh-title\\\">\\n          API\\n        </h2>\\n\\n        <h3 class=\\\"code f5 fw7 mt0 mb1 lh-copy\\\">\\n          GET /3766ff2f-c295-481e-8b78-03d8e04cfc56/notify\\n        </h3>\\n        <p class=\\\"ml2 mv1\\\">\\n          Takes parameters by query string.\\n        </p>\\n\\n        <h3 class=\\\"code f5 fw7 mt2 mb1 lh-copy\\\">\\n          POST /3766ff2f-c295-481e-8b78-03d8e04cfc56/notify\\n        </h3>\\n        <p class=\\\"ml2 mv1\\\">\\n          Takes parameters by JSON body.\\n        </p>\\n\\n        <h4 class=\\\"f5 fw7 mt3 mb1\\\">\\n          Parameters\\n        </h4>\\n        <ul class=\\\"list mv1\\\">\\n          <li>\\n            <span class=\\\"code\\\">title</span>: The title text of the notification.\\n          </li>\\n          <li>\\n            <span class=\\\"code\\\">body</span>: The body text of the notification.\\n          </li>\\n          <li>\\n            <span class=\\\"code\\\">icon</span>: A URL to an icon for the notification.\\n          </li>\\n        </ul>\\n      </section>\\n    </div>\""`;
